### PR TITLE
[ML] Migrate from 2022_12_01_preview API to 2023_02_01_preview API

### PR DESCRIPTION
**Pull request by TonyJ1**
# Description

TODO: Revert the following changes used to test against master MFE:

1. sdk/ml/azure-ai-ml/azure/ai/ml/_ml_client.py `base_url="https://master.api.azureml-test.ms/mferp/managementfrontend"`
2. sdk/ml/azure-ai-ml/tests/conftest.py `add_uri_string_sanitizer()`

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/spark.py [hunk 1]
```diff
@@ -12,11 +12,11 @@
 
 from marshmallow import INCLUDE, Schema
 
-from azure.ai.ml._restclient.v2022_12_01_preview.models import IdentityConfiguration
-from azure.ai.ml._restclient.v2022_12_01_preview.models import JobBase as JobBaseData
-from azure.ai.ml._restclient.v2022_12_01_preview.models import SparkJob as RestSparkJob
-from azure.ai.ml._restclient.v2022_12_01_preview.models import SparkJobEntry as RestSparkJobEntry
-from azure.ai.ml._restclient.v2022_12_01_preview.models import (
+from azure.ai.ml._restclient.v2023_02_01_preview.models import IdentityConfiguration
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/spark.py:15]: Note: may miss some change when this is merged: 
https://github.com/Azure/azure-sdk-for-python/pull/28961

**Comment by novaturient95** [comment]
@TonyJ1 automl jobs were migrated to 12/22 API last week. Will they be migrated to Feb API as part of this PR as well?

**Comment by TonyJ1** [comment]
> @TonyJ1 automl jobs were migrated to 12/22 API last week. Will they be migrated to Feb API as part of this PR as well?

Yes. Please let me know if you find any thing missing for automl.

File: sdk/ml/azure-ai-ml/azure/ai/ml/_ml_client.py [hunk 2]
```diff
@@ -324,7 +324,10 @@ def __init__(
         self._service_client_02_2023_preview = ServiceClient022023Preview(
             credential=self._credential,
             subscription_id=self._operation_scope._subscription_id,
-            base_url=base_url,
+            # base_url=base_url,
```
Comment (TonyJ1) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_ml_client.py:327]: Revert this change once FEB API is available.

File: sdk/ml/azure-ai-ml/tests/conftest.py [hunk 3]
```diff
@@ -98,6 +99,16 @@ def add_sanitizers(test_proxy, fake_datastore_key):
         regex='\\/az-ml-artifacts\\/([^/\\s"]{36})\\/',
         group_for_replace="1",
     )
+    # masks signature in SAS uri
+    add_general_regex_sanitizer(
```
Comment (TonyJ1) [inline sdk/ml/azure-ai-ml/tests/conftest.py:103]: Should this be removed?

File: sdk/ml/azure-ai-ml/tests/conftest.py [hunk 4]
```diff
@@ -98,6 +99,16 @@ def add_sanitizers(test_proxy, fake_datastore_key):
...
+        value="000000000000000000000000000000000000",
+        regex='sig=([^/\\s"]{46,52})',
+        group_for_replace="1",
+    )
+    # TODO: REMOVE before the merge
+    add_uri_string_sanitizer(
```
Comment (TonyJ1) [inline sdk/ml/azure-ai-ml/tests/conftest.py:109]: TODO: Remove before the merge.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/spark.py (hunk 1 above)
Comment (TonyJ1) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/spark.py:15]: Thanks for the note. 
Please let me know if I missed any expected changes.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/workspace.py [hunk 5]
```diff
@@ -8,10 +8,10 @@
 from pathlib import Path
 from typing import IO, AnyStr, Dict, Optional, Union
 
-from azure.ai.ml._restclient.v2023_02_01_preview.models import ManagedServiceIdentity as RestManagedServiceIdentity
-from azure.ai.ml._restclient.v2023_02_01_preview.models import FeatureStoreSettings as RestFeatureStoreSettings
-from azure.ai.ml._restclient.v2023_02_01_preview.models import Workspace as RestWorkspace
-from azure.ai.ml._restclient.v2023_02_01_preview.models import ManagedNetworkSettings as RestManagedNetwork
+from azure.ai.ml._restclient.v2022_12_01_preview.models import ManagedServiceIdentity as RestManagedServiceIdentity
```
Comment (TonyJ1) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/workspace.py:11]: @joshharrin As v2023-02-01_preview doesn't not have few of these APIs reverting to v2022-12.
Please help review and let me know if I miss any changes correspond to [this PR](https://github.com/Azure/azure-sdk-for-python/pull/28879)

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/networking.py [hunk 6]
```diff
@@ -4,7 +4,7 @@
 
 from typing import Any, Dict, Optional
 
-from azure.ai.ml._restclient.v2023_02_01_preview.models import (
+from azure.ai.ml._restclient.v2022_12_01_preview.models import (
```
Comment (TonyJ1) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/networking.py:7]: @joshharrin As v2023-02-01_preview doesn't not have few of these APIs reverting to v2022-12.
Please help review and let me know if I miss any changes correspond to [this PR](https://github.com/Azure/azure-sdk-for-python/pull/28879)

File: sdk/ml/azure-ai-ml/azure/ai/ml/_ml_client.py [hunk 7]
```diff
@@ -263,7 +263,7 @@ def __init__(
             **kwargs,
         )
 
-        self._rp_service_client = ServiceClient122022Preview(
+        self._rp_service_client = ServiceClient022023Preview(
```
Comment (joshharrin) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_ml_client.py:266]: Think you may have missed this one, can we keep ServiceClient122022Preview here please? It is used by WorkspaceOperations and WorkspaceOutboundRuleOperations.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/diagnose.py [hunk 8]
```diff
@@ -3,15 +3,15 @@
 # ---------------------------------------------------------
 from typing import Any, Dict, List, Optional
 
-from azure.ai.ml._restclient.v2022_12_01_preview.models import (
```
Comment (joshharrin) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/diagnose.py:6]: According to previous comment, can workspace continue to use 2022_12_01_preview version?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/workspace_keys.py [hunk 9]
```diff
@@ -4,7 +4,7 @@
 
 from typing import List, Optional
 
-from azure.ai.ml._restclient.v2022_12_01_preview.models import ListWorkspaceKeysResult
+from azure.ai.ml._restclient.v2023_02_01_preview.models import ListWorkspaceKeysResult
```
Comment (joshharrin) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/workspace_keys.py:7]: According to previous comment, can workspace continue to use 2022_12_01_preview version?

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_operations.py [hunk 10]
```diff
@@ -9,8 +9,8 @@
 
 from azure.ai.ml._arm_deployments import ArmDeploymentExecutor
 from azure.ai.ml._arm_deployments.arm_helper import get_template
-from azure.ai.ml._restclient.v2022_12_01_preview import AzureMachineLearningWorkspaces as ServiceClient122022Preview
-from azure.ai.ml._restclient.v2022_12_01_preview.models import (
+from azure.ai.ml._restclient.v2023_02_01_preview import AzureMachineLearningWorkspaces as ServiceClient022023Preview
```
Comment (joshharrin) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_operations.py:12]: According to previous comment, can workspace continue to use 2022_12_01_preview version?

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_outbound_rule_operations.py [hunk 11]
```diff
@@ -3,7 +3,7 @@
 # ---------------------------------------------------------
 
 from typing import Dict
-from azure.ai.ml._restclient.v2022_12_01_preview import AzureMachineLearningWorkspaces as ServiceClient122022Preview
+from azure.ai.ml._restclient.v2023_02_01_preview import AzureMachineLearningWorkspaces as ServiceClient022023Preview
```
Comment (joshharrin) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_outbound_rule_operations.py:6]: According to previous comment, can workspace continue to use 2022_12_01_preview version?

File: sdk/ml/azure-ai-ml/tests/workspace/unittests/test_workspace_operations.py [hunk 12]
```diff
@@ -5,7 +5,7 @@
 from pytest_mock import MockFixture
 
 from azure.ai.ml import MLClient, load_workspace
-from azure.ai.ml._restclient.v2022_12_01_preview.models import (
+from azure.ai.ml._restclient.v2023_02_01_preview.models import (
```
Comment (joshharrin) [inline sdk/ml/azure-ai-ml/tests/workspace/unittests/test_workspace_operations.py:8]: According to previous comment, can workspace continue to use 2022_12_01_preview version?

File: sdk/ml/azure-ai-ml/tests/workspace/unittests/test_workspace_operations.py [hunk 13]
```diff
@@ -31,13 +31,13 @@ def mock_credential() -> Mock:
 @pytest.fixture
 def mock_workspace_operation(
     mock_workspace_scope: OperationScope,
-    mock_aml_services_2022_12_01_preview: Mock,
+    mock_aml_services_2023_02_01_preview: Mock,
```
Comment (joshharrin) [inline sdk/ml/azure-ai-ml/tests/workspace/unittests/test_workspace_operations.py:34]: Also can you change the two in this file as well to be mock_aml_services_2022_12_01_preview

**Review by joshharrin — DISMISSED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/workspace.py (hunk 5 above)
Comment (joshharrin) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/workspace.py]: looks good now

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/networking.py (hunk 6 above)
Comment (joshharrin) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/networking.py]: looks good now

**Review by luigiw — APPROVED** [review]
